build.gradle

```diff
@@ -187,37 +187,18 @@ dependencyCheck {
 dependencyManagement {
   dependencies {
 
-    dependencySet(group: 'org.yaml', version: '1.33') {
+    dependencySet(group: 'org.yaml', version: '2.0') {
       entry 'snakeyaml'
     }
 
-    dependencySet(group: 'org.apache.tomcat.embed', version: '9.0.73') {
-      entry 'tomcat-embed-core'
-      entry 'tomcat-embed-el'
-      entry 'tomcat-embed-websocket'
-    }
-
-    // CVE-2018-10237 - Unbounded memory allocation
-    dependencySet(group: 'com.google.guava', version: '30.0-jre') {
+    dependencySet(group: 'com.google.guava', version: '32.1.2-jre') {
       entry 'guava'
     }
 
     dependencySet(group: 'commons-fileupload', version: '1.5') {
       entry 'commons-fileupload'
     }
 
-    dependencySet(group: 'org.springframework.security', version: '5.7.5') {
-      entry 'spring-security-crypto'
-      entry 'spring-security-core'
```

Comment:
is there a reason for the spring security removal? i know the crypto was removed due to the cve - or were the others no long needed?

Reply:
Entries under `dependencyManagement` are handled by the `io.spring.dependency-management` Gradle plugin, which aims at introducing in Gradle a way of handling dependency versions similar to what Maven used to do.

Essentially, that entry didn't add spring-security but rather overrode its version. By removing it we removed the override, thus allowing the spring version defined in the `org.springframework.boot` plugin to be used instead.

In the past vulnerabilities were found on the previously mapped spring-security version, which forced us to selectively upgrade only spring-security. The updates in this ticket changed the underlying spring version through the `org.springframework.boot` plugin, bumping it up to `2.7.14` , which in turns results in an upgrade of spring-security to `5.7.10`, thus making this override no longer necessary.

Reply:
Perfect thank you Filippo!